107/src/test/java/org/ehcache/jsr107/Eh107JSRTemplateOverridingTest.java

```diff
@@ -0,0 +1,109 @@
+package org.ehcache.jsr107;
+
+import com.pany.domain.Product;
+import com.pany.ehcache.integration.ProductCacheLoaderWriter;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import javax.cache.Caching;
+import javax.cache.configuration.Factory;
+import javax.cache.configuration.MutableConfiguration;
+import javax.cache.expiry.CreatedExpiryPolicy;
+import javax.cache.expiry.Duration;
+import javax.cache.expiry.EternalExpiryPolicy;
+import javax.cache.expiry.ExpiryPolicy;
+import javax.cache.integration.CacheLoader;
+import javax.cache.spi.CachingProvider;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+public class Eh107JSRTemplateOverridingTest {
+
+  private javax.cache.CacheManager cm;
+
+  @Before
+  public void setup() throws Exception {
+    CachingProvider provider = Caching.getCachingProvider();
+    cm = provider.getCacheManager(this.getClass().getResource("/ehcache-107-overriding.xml").toURI(),
+        getClass().getClassLoader());
+  }
+
+  @After
+  public void cleanUp() throws Exception {
+    if (cm != null) {
+      cm.close();
+    }
+  }
+
+  @Test
+  public void testExpiryConfigurationPolicy() {
+    MutableConfiguration<Long, String> conf = new MutableConfiguration<Long, String>();
+    conf.setTypes(Long.class, String.class);
+    javax.cache.Cache<Long, String> c1 = cm.createCache("c1", conf);
+    Eh107CompleteConfiguration<Long, String> cc1 = c1.getConfiguration(Eh107CompleteConfiguration.class);
+
+    assertThat(cc1.getExpiryPolicyFactory(), is(EternalExpiryPolicy.factoryOf()));
+    Factory<ExpiryPolicy> factory = CreatedExpiryPolicy.factoryOf(Duration.ONE_MINUTE);
+    conf.setExpiryPolicyFactory(factory);
+    javax.cache.Cache<Long, String> c2 = cm.createCache("c2", conf);
+    Eh107CompleteConfiguration<Long, String> cc2 = c2.getConfiguration(Eh107CompleteConfiguration.class);
+    assertThat(cc2.getExpiryPolicyFactory(), is(factory));
+  }
+
+
+  @Test
+  public void testCacheLoaderWriterOverriding() {
+    MutableConfiguration<Long, Product> conf = new MutableConfiguration<Long, Product>();
+    conf.setTypes(Long.class, Product.class);
+
+    javax.cache.Cache<Long, Product> c1 = cm.createCache("productCache", conf);
+    ProductCacheLoaderWriter.written.clear();
+    c1.put(1L, new Product(1L));
+    assertThat(ProductCacheLoaderWriter.written.isEmpty(), is(true));
+    cm.destroyCache("productCache");
+
+    conf.setReadThrough(true);
```

Comment:
Are we talking about Modification of mutuable configuration within testcase or within the cachemanger? AFAIK, we cannot modify the configuration within CacheManager as it will cause tck tests fail.